core/templates/dev/head/domain/skill/SkillUpdateService.js

```diff
@@ -0,0 +1,266 @@
+            skill.setDescription(description);
+          }, function(changeDict, skill) {
+            skill.setDescription(oldDescription);
+          });
+      },
+
+      setConceptCardExplanation: function(
+          skill, oldExplanation, newExplanation) {
+        _applySkillContentsPropertyChange(
+          skill, SKILL_CONTENTS_PROPERTY_EXPLANATION,
+          newExplanation, oldExplanation,
+          function(changeDict, skill) {
+            var explanation = _getNewPropertyValueFromChangeDict(changeDict);
+            skill.getConceptCard().setExplanation(explanation);
+          }, function(changeDict, skill) {
+            skill.getConceptCard().setExplanation(oldExplanation);
+          });
+      },
+
+      addWorkedExample: function(skill, newWorkedExample) {
+        var oldWorkedExamples = angular.copy(
+          skill.getConceptCard().getWorkedExamples());
+        var newWorkedExamples = angular.copy(oldWorkedExamples);
+        newWorkedExamples.push(newWorkedExample);
+        _applySkillContentsPropertyChange(
+          skill, SKILL_CONTENTS_PROPERTY_WORKED_EXAMPLES,
+          newWorkedExamples, oldWorkedExamples,
+          function(changeDict, skill) {
+            skill.getConceptCard().setWorkedExamples(newWorkedExamples);
+          }, function(changeDict, skill) {
+            skill.getConceptCard().setWorkedExamples(oldWorkedExamples);
+          });
+      },
+
+      deleteWorkedExample: function(skill, index) {
+        var oldWorkedExamples = angular.copy(
+          skill.getConceptCard().getWorkedExamples());
+        var newWorkedExamples = angular.copy(oldWorkedExamples);
+        newWorkedExamples.splice(index, 1);
+        _applySkillContentsPropertyChange(
+          skill, SKILL_CONTENTS_PROPERTY_WORKED_EXAMPLES,
+          newWorkedExamples, oldWorkedExamples,
+          function(changeDict, skill) {
+            skill.getConceptCard().setWorkedExamples(newWorkedExamples);
+          }, function(changeDict, skill) {
+            skill.getConceptCard().setWorkedExamples(oldWorkedExamples);
+          });
+      },
+
+      updateWorkedExample: function(
+          skill, workedExampleIndex, newWorkedExample) {
+        var oldWorkedExamples = angular.copy(
+          skill.getConceptCard().getWorkedExamples());
+        var newWorkedExamples = angular.copy(oldWorkedExamples);
+        newWorkedExamples[workedExampleIndex] = newWorkedExample;
+        _applySkillContentsPropertyChange(
+          skill, SKILL_CONTENTS_PROPERTY_WORKED_EXAMPLES,
+          newWorkedExamples, oldWorkedExamples,
+          function(changeDict, skill) {
+            var newWorkedExamples =
+              _getNewPropertyValueFromChangeDict(changeDict);
+            skill.getConceptCard().setWorkedExamples(newWorkedExamples);
+          }, function(changeDict, skill) {
+            skill.getConceptCard().setWorkedExamples(oldWorkedExamples);
+          });
+      },
+
+      updateWorkedExamples: function(
```

Comment:
Can't this be achieved by repeatedly calling the above function? or is there a case where this would be advantageous?